packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.test.js

```diff
@@ -849,3 +851,391 @@ describe('multi column layout test cases', () => {
+      { top: 0, left: 505, width: 236, height: 274 },
+      { top: 0, left: 757, width: 236, height: 303 },
+      { top: 0, left: 1009, width: 236, height: 475 },
+      { top: 0, left: 1261, width: 236, height: 496 },
+      { top: 0, left: 1513, width: 236, height: 177 },
+      { top: 0, left: 1765, width: 236, height: 440 },
+      { top: 0, left: 2017, width: 236, height: 497 },
+      { top: 193, left: 1513, width: 236, height: 430 },
+      { top: 290, left: 505, width: 236, height: 409 },
+      { top: 319, left: 757, width: 236, height: 452 },
+      { top: 397, left: 253, width: 236, height: 458 },
+      { top: 456, left: 1765, width: 236, height: 510 },
+      { top: 491, left: 1009, width: 236, height: 336 },
+      { top: 715, left: 505, width: 236, height: 322 },
+      { top: 787, left: 757, width: 236, height: 417 },
+    ]);
+
+    const heights = initializeHeightsArray({
+      centerOffset: 1,
+      columnCount: 9,
+      columnWidthAndGutter: columnWidth + gutter,
+      gutter,
+      items,
+      positionCache,
+    });
+
+    expect(heights.length).toEqual(9);
+    expect(heights).toEqual([924, 871, 1053, 1220, 843, 1421, 1421, 982, 829]);
+
+    const additionalItems = [
+      {
+        'name': 'Pin: 23',
+        'height': 428,
+      },
+      {
+        'name': 'Pin: 24',
+        'height': 340,
+      },
+      {
+        'name': 'Pin: 25',
+        'height': 458,
+      },
+      {
+        'name': 'Pin: 26',
+        'height': 475,
+      },
+      {
+        'name': 'Pin: 27',
+        'height': 303,
+      },
+      {
+        'name': 'Pin: 28',
+        'height': 519,
+      },
+      {
+        'name': 'Pin: 29',
+        'height': 440,
+      },
+      {
+        'name': 'Pin: 30',
+        'height': 391,
+      },
+      {
+        'name': 'Pin: 31',
+        'height': 475,
+      },
+      {
+        'name': 'Pin: 32',
+        'height': 458,
+      },
+      {
+        'name': 'Pin: 33',
+        'height': 292,
+      },
+      {
+        'name': 'Pin: 34',
+        'height': 215,
+      },
+      {
+        'name': 'Pin: 35',
+        'height': 400,
+      },
+      {
+        'name': 'Pin: 36',
+        'height': 153,
+      },
+    ];
+    additionalItems.forEach((item) => {
+      measurementStore.set(item, item.height);
+    });
+
+    layout(items.concat(additionalItems));
+
+    const newPositions = additionalItems.map((item) => positionCache.get(item));
+
+    const newPositionsByColumns = newPositions
+      // $FlowFixMe[missing-local-annot]
+      .reduce((acc, position) => {
+        const column = Math.floor((position?.left ?? 0) / (columnWidth + gutter));
+        if (!acc[column]) {
+          acc[column] = [];
+        }
+        acc[column].push(position);
+        return acc;
+      }, [])
+      .map((column) => column.sort((a, b) => (a?.top ?? 0) - (b?.top ?? 0)));
+
+    const newTops = newPositionsByColumns.map((column) => column[0].top);
+    expect(newTops).toEqual(heights);
```

Comment:
I see that you are verifying that after adding new items the new items start were the previous items ended (the previous heights), just not super sure why this would be necesary, isn't the issue present on the first heightsInitialize? Maybe a little bit of description here could help

Reply:
done. lmk if this is clearer. basically what i'm doing here is:
- layout
- validate heights based on initial items are correct
- layout
- validate positions are correct (based on previously validated heights)